### ISSUE TITLE
Forms initialized with existing array items do not display all items

### ISSUE THREAD

**Issue by tmoc** [json-schema-form/angular-schema-form#287]
When initializing a form using existing data (such as from a database), everything works as expected except for arrays. Only one array element is displayed and attempting to add more causes odd behavior where it isn't clear what the actual state of the array is from the form.

**Comment by davidlgj** [comment]
I did a simple test to reproduce, but that worked fine for me (with two elements in an array with objects). Could you give me an example schema,form definition and model state that fails for you?

**Comment by tmoc** [comment]
<pre>
  $scope.editSchema = {
    type: 'object',
    properties: {
      _id: {title: '_id*', type: 'string', pattern: '^[0-9a-fA-F]{24}$', readonly: true},
      title: {type: 'string'},
      sku: {type: 'string'},
      type: {type: 'string'},
      price: {type: 'string'},
      chargifyHandle: {type: 'string'},
      headline: {type: 'string'},
      description: {type: 'string'},
      images: {
        type: 'array',
        items: {
          type: 'string'
        }
      }
    },
    required: ['_id']
  };

  $scope.sfEditForm = [
    {key: '_id'},
    {key: 'title'},
    {key: 'sku'},
    {key: 'type'},
    {key: 'price'},
    {key: 'chargifyHandle'},
    {key: 'headline'},
    {key: 'description'},
    {key: 'images'},
    {
      type: 'actions',
      items: [
        {type: 'submit', style: 'btn-primary', title: 'Update'},
        {type: 'button', style: 'btn-danger', title: 'Delete', onClick: 'confirmDelete()'}
      ]
    }
  ];

Object {_id: "54fde9551b94957698f16868", _documentCreatedAt: "2015-03-09T18:41:25.808Z", _documentUpdatedAt: "2015-03-09T18:41:25.808Z", title: "test", sku: "test"…}$$hashKey: "05J"__v: 0_documentCreatedAt: "2015-03-09T18:41:25.808Z"_documentUpdatedAt: "2015-03-09T18:41:25.808Z"_id: "54fde9551b94957698f16868"chargifyHandle: "test"images: Array[2]0: "test1"1: "test2"length: 2__proto__: Array[0]price: "test"sku: "test"title: "test"type: "test"__proto__: Object

</pre>

**Comment by davidlgj** [comment]
Ah, I see, I just tested with an object as the base type in the array, not an array of simple types... 
Great that you found a workaround! I'll add this bug to the todo :)

**Comment by tmoc** [comment]
Thanks. Great library btw.

**Comment by Li-ReDBox** [comment]
I had similar issue but am not sure if it is exactly the same problem. I want to display an array of objects. When I set the model by a variable inside the view function, it displays correct number (more than 2) of array items. But when it loads from server, like this:
```javascript
  resource.get({datatype:$routeParams.datatype}, function(res){
    $scope.model = res.content;
    console.log($scope.model);
    $scope.$broadcast('schemaFromRedraw');
  });
```
it only shows two items in the array and broadcasting the redraw event as above does not help. By clicking "Add" button of this array it will load the next item. I printed `model` on page which had all items in the array. So I guess it could relate to redraw somehow.

Thanks for your good work.

**Comment by davidlgj** [comment]
@Li-ReDBox yeah, it sounds like it might be the same bug :(

**Comment by Li-ReDBox** [comment]
@davidlgj I spent some more time on this issue again on my end. I first realized that the event name of redraw I copied from above was not correct which should be `schemaFormRedraw` not `schemaFromRedraw`. Then I noticed my local version of schemaform was 0.7.3 which was behind the current release. So I updated it to 0.7.13. After that, so far it has been working as expected and I have not force form to redraw.

I am sorry for giving you wrong information. Once again, thank you for your good work.

**Comment by shawnss** [comment]
I am having the same issue with my own addon widget. I have put together kendo (telerik) date picker, time picker and datetime picker. After updating the model with existing data, the date picker was updated with the value but the timepicker and datetime picker were not. All other form controls were updated. I just saw this issue and added schemaFormRedraw broadcast after setting the model and all is working now. Not sure if this is the right way of fixing it, but it works now.
